processor/metricstransformprocessor/config.go

```diff
@@ -85,6 +85,11 @@ type Operation struct {
 
 	// ValueActions is a list of renaming actions for label values.
 	ValueActions []ValueAction `mapstructure:"value_actions"`
+
+	// fields that will be filled later for optimized opertaions processing
```

Comment:
I actually started by modifying the internal config but then I noticed I had to recreate the Config for the internal config, and I also had to traverse through everything to copy over. Since this config is a rather deep one, so I didn't like how I had to traverse through everything just to copy things over. I am now using an implementation that I consider to be at a good balance where I am reusing part of the initial config. Please let me know how it looks.

Reply:
You raise a good point about wanting to avoid a bunch of mapping code / duplication. There is often a tradeoff between duplication and isolation / readability. While it's annoying to write mapping code, it helps to simplify the more complex part of the code here imo. Being able to freely change your internal implementation separately from the structure of the config is fairly valuable for this processor.

Note a similar example where the [attribute processor](https://github.com/open-telemetry/opentelemetry-collector/blob/cae568d276ed9d28a498281d35abf48777ad9460/internal/processor/attraction/attraction.go#L138) does this. It's slightly simpler since there is one less level of nesting than you need here, but not significantly.

One possible option to avoid having to copy all the fields would be to do something like:
- Rename existing Transform struct to TransformConfig
- Rename existing Operation struct to OperationConfig
- Add:
```go
type Transform struct {
  transform *TransformConfig
  operations []Operation
}

type Operation struct {
  operation *OperationConfig

  valueActionsMapping map[string]string
  labelSetMap         map[string]bool
  aggregatedValuesSet map[string]bool
}
```

But that's probably not as good as creating entirely new structs imo.

There's no perfect answer, so up to you what you want to do, but I would definitely want to avoid:
- Having these internal mappings in the same struct as the config
- Exporting these fields

Reply:
Checkout my new commits. I did something very similar to your suggestion. Let me know if it looks good.